xdsl/transforms/canonicalization_patterns/cf.py

```diff
@@ -142,9 +142,12 @@ def match_and_rewrite(self, op: cf.ConditionalBranchOp, rewriter: PatternRewrite
         # Check if cond operand is constant
         cond = const_evaluate_operand(op.cond)
 
-        if cond == 1:
+        if cond is None:
+            return
+
+        if cond:
             rewriter.replace_matched_op(cf.BranchOp(op.then_block, *op.then_arguments))
-        elif cond == 0:
+        else:
             rewriter.replace_matched_op(cf.BranchOp(op.else_block, *op.else_arguments))
```

Comment:
I can't work out how to attach the comment in the correct place but line 374 of this file will now be incorrect